http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1292,4 +1320,43 @@ public RequestDistribution(HttpClient httpClient, Processor processor)
             });
         }
     }
+
+    // By wrapping HttpClient, we are able to substitute the underlying AuthenticationStore
+    // with a more efficient one.
+    private static class WrappedHttpClient
+            extends HttpClient
+    {
+        private final AuthenticationStore authenticationStore;
+        private final SpnegoAuthentication spnego;
+
+        public WrappedHttpClient(KerberosConfig kerberosConfig, HttpClientConfig config, HttpClientTransport transport, SslContextFactory sslContextFactory)
```

Comment:
Maybe name this something like `SpnegoHttpClient` or `SecureHttpClient`... wrapped is just really generic